firstvoices/backend/views/import_job_views.py

```diff
@@ -159,3 +193,49 @@ def confirm(self, request, site_slug=None, pk=None):
         return Response(
             serializer.data, status=status.HTTP_202_ACCEPTED, headers=headers
         )
+
+    @action(detail=True, methods=["post"])
+    def validate(self, request, site_slug=None, pk=None):
+        """
+        Method to start the validation process on a given import-job.
+        """
+        site = self.get_validated_site()
+        import_job_id = self.kwargs["pk"]
+        logger = logging.getLogger(__name__)
+
+        # Verify that no other jobs are started or queued for the same site
+        existing_incomplete_jobs = get_import_jobs_queued_or_running(
+            site, import_job_id
+        )
+
+        if len(existing_incomplete_jobs):
+            logger.error(
+                "There is at least 1 job on this site that is already running or queued to run soon. "
+                "Please wait for it to finish before starting a new one."
+            )
+            logger.info(ASYNC_TASK_END_TEMPLATE)
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+
+        # Edge case check to prevent running validation on the current job
+        # if its already running or queued for dry run
+        curr_job = ImportJob.objects.filter(id=import_job_id)[0]
+        if curr_job.validation_status in [JobStatus.ACCEPTED, JobStatus.STARTED]:
+            logger.error(
+                "The specified job is already running or queued. "
```

Comment:
Same issue here-- user messages need to go in the Response object. The easiest way is usually to raise a ValidationError rather than manually constructing the response. Something like `raise raise ValidationError("The specified job is already running etc etc...")`